x/exchange/types/escrow.go

```diff
@@ -49,9 +49,8 @@ func (e *Escrow) Pays(addr sdk.AccAddress) sdk.Coins {
 
 func (e *Escrow) Receives(addr sdk.AccAddress) sdk.Coins {
 	var receives sdk.Coins
-	for _, decCoin := range e.deltas[addr.String()] {
-		if decCoin.IsPositive() {
-			coin, _ := decCoin.TruncateDecimal()
+	for _, coin := range e.deltas[addr.String()] {
+		if coin.IsPositive() {
 			receives = receives.Add(coin)
 		}
```

Comment:
What about the case of IsNegative()? If it is not allowed, the error handling (or sanity check) is needed.

Reply:
Ditto